### DISCUSSION TITLE
setInterval method in Timer Class

### DISCUSSION THREAD

**Discussion by yzPeedro** [reactphp/reactphp#490] [Ideas]
I've been wondering if we could change the interval of Period Timer inside of loop, for example

```php
$interval = 0;
Loop::addPeriodicTimer($interval, function($timer) use(&$interval) {
    $interval++;
    echo "TICK: " . $interval;
    $timer->setInterval($interval); //new method
});
```

That way we could change the loop time within itself, or even on another occasion that was necessary.

**Comment by SimonFrings** [comment]
Hey @yzPeedro, thanks for bringing up your idea, always happy about input :+1:

Seems like a useful function to suggest, but I think this is already possible in a different way. For example, you could create a new timer instance with the `addTimer()` function and after the timer runs out you increment the `$interval`, cancel the current timer, and create a new timer instance with the incremented `$interval`. This timer could then have the same logic and so on. This way we would also have only one timer instance at a time, because the old ones get cancelled, which matches your suggestion.

With this information I would pass the following question back to you: Is something gained by adding the function you suggested instead of using the described workaround?

I'm interested to hear you thoughts on this!

**Comment by yzPeedro** [comment]
I thought of this solution using ```addTimer```, but the specific problem that sounds bad to me is the pollution of the code, since I would technically be executing two actions that are almost the same, if the ```setInterval``` method existed in the Timer class, I could manipulate the Timer directly in the function I'm executing, without putting external logic inside the function I want to loop, I think it would make even more sematic sense in terms of the code, it seems wrong to write two ```addTimer``` when I could just write one and change the interval of it as per code logic. It seems less problematic to write a single Timer and change its interval than to write two Timers to change the interval of one.

**Reply by SimonFrings** [reply]
I can see where you're coming from and I think you can achieve the same outcome as your example above by using some kind of logic that calls the same timer function a bunch of times, each time with a different `$interval`. You could achieve that by wrapping the timer function into some sort of loop or function which can be called if needed.

We could implement this sort of behavior but I also see the amount of tests and edge cases behind this. In my eyes my suggested workarounds would solve this case, this is why I'm not convinced that this is something the project needs, but maybe I don't see the whole picture here. What exactly are you trying to achieve with this mechanism, what's your use case?

**Reply by yzPeedro** [reply]
I believe that the ```addTime``` solution is great for my case, I needed a loop that would change the time based on the time of the song that was playing on my spotify, so I would create a handle for whenever the music changes, based on the tempo it is playing, so the loop would have the same tempo as the music, but I believe this ```addTimer``` example works fine.  I suggested the option of addInterval because it would make the work of the handle a lot easier, since with just one line of code inside the Loop I could already change its time.
